aws-sns-topicpolicy/src/main/java/software/amazon/sns/topicpolicy/CreateHandler.java

```diff
@@ -1,77 +1,71 @@
-     *            {@link ResourceHandlerRequest}
-     * @param callbackContext
-     *            {@link CallbackContext}
-     * @return {@link ProgressEvent}
-     */
+        logger.log(String.format("[StackId: %s, ClientRequestToken: %s] Calling Create SNS TopicPolicy", request.getStackId(), request.getClientRequestToken()));
+
+        return ProgressEvent.progress(resourceModel, callbackContext)
+                .then(progress -> Create(proxy, proxyClient, request, progress, logger))
+                .then(progress -> ProgressEvent.defaultSuccessHandler(null));
+    }
 
-    private ProgressEvent<ResourceModel, CallbackContext> initCallbackContextAndPrimaryIdentifier(
+    protected ProgressEvent<ResourceModel, CallbackContext> Create(
             final AmazonWebServicesClientProxy proxy,
             final ProxyClient<SnsClient> proxyClient,
             final ResourceHandlerRequest<ResourceModel> request,
-            final CallbackContext callbackContext,
-            ProgressEvent<ResourceModel, CallbackContext> progress) {
-
-        final CallbackContext currentContext = callbackContext == null ? CallbackContext
-                .builder()
-                .build() : callbackContext;
+            ProgressEvent<ResourceModel, CallbackContext> progress,
+            final Logger logger) {
         final ResourceModel model = request.getDesiredResourceState();
-        // setting up primary id if not provided
-        if (StringUtils.isNullOrEmpty(model.getId())) {
-            final String Id = IdentifierUtils.generateResourceIdentifier(
-                    request.getLogicalResourceIdentifier() == null ? "SnsTopicPolicy"
-                            : request.getLogicalResourceIdentifier(),
-                    request.getClientRequestToken(),
-                    MAX_LENGTH_SNS_TOPICPOLICY_ID);
-            model.setId(Id.toLowerCase());
+        final CallbackContext callbackContext = progress.getCallbackContext();
+        final String policy = getPolicyDocument(request);
+        List<String> topics = model.getTopics();
+        for (final String topicArn : topics) {
+            final ProgressEvent<ResourceModel, CallbackContext> progressEvent = proxy
+                    .initiate("AWS-SNS-TopicPolicy::Create", proxyClient, model, callbackContext)
+                    .translateToServiceRequest((resourceModel) -> Translator.translateToRequest(topicArn, policy))
+                    .makeServiceCall((awsRequest, client) -> {
+                        SetTopicAttributesResponse response = proxyClient.injectCredentialsAndInvokeV2(awsRequest, client.client()::setTopicAttributes);
+                        logger.log(String.format("Resource Created in StackId: %s", request.getStackId()));
+                        return response;
+                    })
+                    .handleError((awsRequest, exception, client, rModel, context) -> handleError(awsRequest, exception, client, rModel, context))
+                    .success();
```

Comment:
need to add this 30 second propagation delay: 
https://code.amazon.com/packages/AWS21Workflows/blobs/mainline/--/src/com/amazon/aws21/workflows/resources/sns/topicPolicy/TopicPolicyCreateWorkflowImpl.java#:~:text=waitFor%20%3D%20createHandler,10%20years%20ago

Example of a propagation delay: https://code.amazon.com/packages/AWSCloudFormationResourceProvidersEC2VPNGateway-TB/blobs/mainline/--/vpngateway/src/main/java/software/amazon/ec2/vpngateway/CreateHandler.java#:~:text=if%20(progress,%7D)